api/v1/agentconfig_types.go

```diff
@@ -207,12 +207,20 @@ func init() {
 	SchemeBuilder.Register(&AgentConfig{}, &AgentConfigList{})
 }
 
+type PluginsSpec struct {
+	// SchemaVersion is the version of the plugins configuration state schema.
+	SchemaVersion string `json:"schemaVersion,omitempty" mapstructure:"schemaVersion,omitempty"`
+
+	// Configs is a map of plugin configuration using plugin name as the key.
+	Configs map[string]Plugin `json:"configs,omitempty" mapstructure:"configs,omitempty"`
```

Comment:
I'm a bit confused about why this representation of the document uses `configs` instead of `plugins`?

Reply:
```yaml
spec:
  plugins:
    schemaVersion: 1.0.0
    configs:
      kubernetes: {}
```
This is what the schema looks like with the current implementation
If we change from `Configs` to `Plugins`, I find it a bit confusing to have `Plugins` under `Spec.Plugins`

Reply:
That's a very good point. I changed it based on your suggestions